apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette-contentlets/dot-palette-contentlets.component.ts

```diff
@@ -0,0 +1,125 @@
+import {
+    Component,
+    EventEmitter,
+    Input,
+    OnChanges,
+    Output,
+    SimpleChanges,
+    ViewChild
+} from '@angular/core';
+import { DotContentletEditorService } from '@dotcms/app/view/components/dot-contentlet-editor/services/dot-contentlet-editor.service';
+import { DotCMSContentlet, DotCMSContentType } from '@dotcms/dotcms-models';
+import { DotPaginatorESContentService } from '@services/dot-paginator-es-content/dot-paginator-es-content.service';
+import { PaginatorService } from '@services/paginator';
+import { LazyLoadEvent } from 'primeng/api';
+import { take } from 'rxjs/operators';
+import { DotPaletteInputFilterComponent } from '../dot-palette-input-filter/dot-palette-input-filter.component';
+
+@Component({
+    selector: 'dot-palette-contentlets',
+    templateUrl: './dot-palette-contentlets.component.html',
+    styleUrls: ['./dot-palette-contentlets.component.scss']
+})
+export class DotPaletteContentletsComponent implements OnChanges {
+    @ViewChild('filterInput', { static: true })
+    filterInput: DotPaletteInputFilterComponent;
+
+    @Input() contentTypeVariable: string;
+    @Output() hide = new EventEmitter();
+    items: DotCMSContentlet[] | DotCMSContentType[];
+    isFormContentType: boolean;
+    hideNoResults = true;
+
+    constructor(
+        public paginatorESService: DotPaginatorESContentService,
+        public paginationService: PaginatorService,
+        private dotContentletEditorService: DotContentletEditorService
+    ) {}
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes?.contentTypeVariable?.currentValue) {
+            this.isFormContentType = changes?.contentTypeVariable?.currentValue === 'forms';
+
+            if (this.isFormContentType) {
+                this.paginationService.url = `v1/contenttype`;
+                this.paginationService.paginationPerPage = 15;
+                this.paginationService.sortField = 'modDate';
+                this.paginationService.setExtraParams('type', 'Form');
+                this.paginationService.sortOrder = 1;
```

Comment:
If we are only using the `paginationService` for the forms we don't need to set all this every time. This can be set once in the `ngOnInit`.

Reply:
`DotPaletteContentletsComponent` is loaded once and then its content is changed during user's behaviours so it needs to be on the OnChange method

Reply:
yes, but due to race conditions those params are rewritten with other values from other components